src/reducers/windowHandler.js

```diff
@@ -41,7 +41,16 @@ const initialState = {
     latestNewDocument: null,
     viewId: null,
     selected: [],
-    selectedWindowType: null
+    selections: {},
+};
+
+export const NO_SELECTION = [];
+export const getSelection = ({ state, windowType, viewId }) => {
```

Comment:
Ideally, `windowType` and `viewId` should be managed and taken from the store directly. Currently `viewId` is managed in `DocumentList`'s internal state which is hard to share between all components reliably. When it's only passed down as a prop, we can't share it with sibling or parent components and lose the overview quick.

Reply:
The problem is if we keep `windowType` and `viewId` in store, there still are different `windowType`s and `viewId`s per `DocumentList`. And there can be 4 different `DocumentList`s active at the same time: modal and non-modal, and additionally as `includedView` each.

But you are right, `viewId` can not be shared with components other than its children. Actually this line (48) https://github.com/metasfresh/metasfresh-webui-frontend/blob/a49cbf8d11878b885d74a1e833c5e4865b51d6b8/src/components/app/DocumentList.js#L44-L50 uses `defaultViewId` from `props` instead of `state`, because there is no way to access `state` or `context` in `connect()`...

---

This is confusing, but not because of this `connect()` but because of the existence of `defaultViewId`. The `viewId` from `state` is only changed here:
- [x] In `constructor()`: equals to `defaultViewId`:
https://github.com/metasfresh/metasfresh-webui-frontend/blob/a49cbf8d11878b885d74a1e833c5e4865b51d6b8/src/components/app/DocumentList.js#L65-L69
- [x] In `componentWillReceiveProps()`: `viewId` was retrieved from `state`
https://github.com/metasfresh/metasfresh-webui-frontend/blob/a49cbf8d11878b885d74a1e833c5e4865b51d6b8/src/components/app/DocumentList.js#L123-L138
- [x] In `componentWillReceiveProps()` as well: `defaultViewId` from `nextProps`, so `connect()` will update its selection
https://github.com/metasfresh/metasfresh-webui-frontend/blob/a49cbf8d11878b885d74a1e833c5e4865b51d6b8/src/components/app/DocumentList.js#L153-L157
- [ ] In `createView()`: This function is called, if no `viewId` is given, in which case `connect()` can not retrieve any selections anyway. I think this point might be an issue, as the selection will be invalid in this case!
https://github.com/metasfresh/metasfresh-webui-frontend/blob/a49cbf8d11878b885d74a1e833c5e4865b51d6b8/src/components/app/DocumentList.js#L314-L324